src/berry_mill/builder.py

```diff
@@ -0,0 +1,198 @@
+        if not repo_key_path:
+            raise Exception("Key path not defined")
+        return "file:///" + \
+               quote( \
+               os.path.join( \
+               os.path.basename(self._boxtmpkeydir), os.path.basename(repo_key_path)))
+
+
+    def _write_repokeys_box(self, repos:Dict[str, Dict[str, str]]) -> None:
+        """
+        Write repo keys from the self._tmp dir to the boxroot, so kiwi box can access the keys
+        It expects a file:// uri to be present in repos[reponame][key]
+        """
+        dst = self._boxtmpkeydir
+        if not dst:
+            raise Exception("ERROR: Boxroot directory is not defined")
+
+        for reponame in repos:
+            k = repos.get(reponame, {}).get("key")
+            parsed_url = urlparse(k)
+            try:
+                shutil.copy(parsed_url.path, dst)
+            except Exception as exc:
+                print(f"ERROR: Failure while trying to copying the keyfile at {parsed_url.path}")
+                print(exc)
+                return
+            repos.get(reponame, {})["key"] = self._get_relative_file_uri(parsed_url.path)
+
+    def process(self) -> None:
+        """
+        Run builder. It supposed to be already within that directory (os.chdir).
+        Directory is changed by the parent caller of the KiwiBuilder class.
+        """
+        # options that are solely accepted by kiwi-ng
+        kiwi_options = self._kiwi_options
+        # options, solely accepted by box-build plugin
+        box_options:List[str] = ["--box","ubuntu"]
+
+        if self._kiwiparams.get("debug", False):
+            box_options.append("--box-debug")
+
+        if self._params.get("cpu"):
+            box_options = ["--cpu", self._params.get("cpu")] + box_options
+
+        if self._params.get("box_memory"):
+            box_options += ["--box-memory", self._params.get("box_memory")]
+
+        if machine() == "aarch64":
+            box_options += ["--machine", "virt"]
+
+        allow_no_accel:bool = True
+        # TODO: When using cross, e.g. cpu param needs to be disabled
+        if self._params.get("cross") and machine() == "x86_64":
+            box_options += ["--aarch64", "--cpu", "cortex-a57", "--machine", "virt", "--no-accel"]
+            kiwi_options += ["--target-arch", "aarch64"]
+            allow_no_accel = False
+
+        if self._params.get("accel", False) and allow_no_accel:
+            box_options.append("--no-accel")
+
+        try:
+            config_tree = etree.parse(f"{self._appliance_descr}")
+        except Exception as err:
+            print("ERROR: Failure {} while parsing appliance description".format(err))
+            return
+
+        try:
+            image_name:list = config_tree.xpath("//image/@name")
+        except Exception as err:
+            print("ERROR: Failure {} while trying to extract image name", err)
+            return
+
+        target_dir = os.path.join(
+            self._params.get("target_dir","/tmp"),\
```

Comment:
Here `/tmp` is hardcoded. I would prefer putting it into a class variable instead. And second, `/tmp` is not always the best choice as it might be too small.

Reply:
I will do it differently:
I force the user to provide a `--target-dir` argument when invoking `berrymill build`

Here I will just assert that a target dir key is existent